.drone.yml

```diff
@@ -92,44 +128,112 @@ steps:
       source: dist/**/*
       strip_prefix: dist/
       target: dev/main
-    when:
-      event: push
-      branch:
-        - main
     depends_on:
-      - build project
+      - git tag push
+
+---
+kind: pipeline
+type: kubernetes
+name: callisto-ui-test-deploy
+
+platform:
+  os: linux
+  arch: amd64
+
+depends_on:
+  - callisto-ui-dev-deploy
+
+trigger:
+  event:
+    - promote
```

Comment:
Does this alone prevent us from being able to promote a build from any branch that isn't main?
Do we need something else to only allow promoting once it's pushed to main, or was that to come separately?

Reply:
Good shout. I was going to look into ways to protect only promoting from the main branch next, but I think the options might be limited with Drone...

Reply:
Ah ok, I just tested on the timecard one, if you include an event here for push for 'main' branch and then attempt to promote a from a non main branch although it will still run, it will not run the test depl pipeline.
So that's a partial solution in that it prevents the main stuff but unfortunately still allows us to trigger the promotion.

I'll resolve now anyway, perhaps we can discuss afterwards..